libid/complex.cpp

```diff
@@ -0,0 +1,536 @@
+    }
+
+Complex Complex::operator^(double & expon)
+    {
+    Complex	temp, temp1;
+
+    temp.x = x;
+    temp.y = y;
+    temp1 = temp.CLog()*expon;
+//    x = temp1.x;
+//    y = temp1.y;
+    temp = temp1.CExp();
+//    temp.x = x;
+//    temp.y = y;
+//    temp = CExp(CLog(temp)*expon);
+    return temp;			
+    }
+
+Complex Complex::operator^(Complex & expon)
+    {
+    Complex	temp, temp1;
+
+    temp.x = x;
+    temp.y = y;
+    temp1 = temp.CLog()*expon;
+    temp = temp1.CExp();
+    //    temp.x = x;
+    //    temp.y = y;
+    //    temp = CExp(CLog(temp)*expon);
+    return temp;
+    }
+
+Complex Complex::operator++(void)
+    {
+    x+=1;
+    return *this;
+    }
+// end new stuff
+
+Complex Complex::operator+(const Complex & Cmplx1)	// complex add Operator
+    {
+    Complex	temp;
+
+    temp.x = x + Cmplx1.x;
+    temp.y = y + Cmplx1.y;
+    return  temp;
+    }
+
+Complex Complex::operator+(const double & Sum)	// complex add double Operator
+    {
+    Complex	temp;
+
+    temp.x = x + Sum;
+    temp.y = y;
+    return  temp;
+    }
+
+Complex Complex::operator-(const Complex & Cmplx1)	// complex subtract Operator
+    {
+    Complex	temp;
+
+    temp.x = x - Cmplx1.x;
+    temp.y = y - Cmplx1.y;
+    return  temp;
+    }
+
+Complex Complex::operator-(const double & Difference)	// complex subtract double Operator
+    {
+    Complex	temp;
+
+    temp.x = x - Difference;
+    temp.y = y;
+    return  temp;
+    }
+
+Complex Complex::operator-(void)		    // unary minus
+    {
+    Complex	temp;
+
+    temp.x = -x;
+    temp.y = -y;
+    return  temp;
+    }
+
+Complex Complex::operator*(const Complex & Cmplx1)	// complex multiply Operator
+    {
+    Complex	temp;
+
+    temp.x = x * Cmplx1.x - y * Cmplx1.y;
+    temp.y = y * Cmplx1.x + x * Cmplx1.y;
+    return  temp;
+    }
+
+Complex Complex::operator*(const double & Multiplier)	// complex multiply by double Operator
+    {
+    Complex	temp;
+
+    temp.x = x * Multiplier;
+    temp.y = y * Multiplier;
+    return  temp;
+    }
+
+Complex Complex::operator/(const Complex & Cmplx1)	// complex divide Operator
+    {
+    double	d;
+    Complex	temp;
+
+    if (Cmplx1.x == 0.0 && Cmplx1.y == 0.0)
+	d = zerotol;					// prevent divide by zero
+    else
+	d = Cmplx1.x * Cmplx1.x + Cmplx1.y * Cmplx1.y;
+    temp.x = (x * Cmplx1.x + y * Cmplx1.y) / d;
+    temp.y = (y * Cmplx1.x - x * Cmplx1.y) / d;
+    return  temp;
+    }
+
+Complex Complex::operator/(const double & divisor)	// complex divide by double Operator
+    {
+    Complex	temp;
+    double  d;
+
+    if (divisor == 0.0)
+	d = zerotol;					// prevent divide by zero
+    else
+	d = divisor;
+    temp.x = x / d;
+    temp.y = y / d;
+    return  temp;
+    }
+
+/**************************************************************************
+	sin(x+iy)  = sin(x)cosh(y) + icos(x)sinh(y)
+***************************************************************************/
+
+Complex	Complex::CSin()
```

Comment:
This should be a free function instead of a member function and just named `sin` instead of `CSin`.  Same for the other trig functions.

Always prefer a free (e.g. plain) function over a member function when possible.